database-commons/src/main/java/io/cdap/plugin/db/sink/ETLDBOutputFormat.java

```diff
@@ -98,23 +183,15 @@ public void close(TaskAttemptContext context) throws IOException {
             if (!emptyData) {
               getStatement().executeBatch();
             }
-            getConnection().commit();
           } catch (SQLException e) {
-            try {
-              getConnection().rollback();
-            } catch (SQLException ex) {
-              LOG.warn(StringUtils.stringifyException(ex));
-            }
             throw new IOException(e);
           } finally {
             try {
               getStatement().close();
-              getConnection().close();
             } catch (SQLException ex) {
               throw new IOException(ex);
             }
           }
-
           try {
             DriverManager.deregisterDriver(driverShim);
```

Comment:
After this call, driver may invalidate the connection, see java.sql.DriverAction#deregister:
"it is implementation specific as to whether the connections are closed or allowed to continue."

Reply:
Discussed offline, DriverAction is not used in CDAP